tests/scripts/check-names.sh

```diff
@@ -28,8 +28,8 @@ EOF
     exit
 fi
 
-if grep --version|head -n1|grep GNU >/dev/null; then :; else
-    echo "This script requires GNU grep.">&2
+if grep --version|head -n1|grep -E "GNU|FreeBSD" >/dev/null; then :; else
```

Comment:
The script actually works with any grep implementation. This check was added in 71ebc58932b78e412f1ba8693da5a8bf7e506f9a at the same time as a use of `\|` in BRE which is not portable, but this use of `\|` was removed later (and it works on modern *BSD anyway).

There are two remaining portability issues:

* Many uses of `egrep` (a nonstandard but common alias for `grep -E`), which is not guaranteed to exist, but exists on modern *BSD anyway.
* One use of `\|` in a BRE, but it's under `if uname | grep -F Linux`.

Therefore, please just remove the version check altogether. Optionally fix the places where `grep -E` should be used, but that's not important.